src/components/Email.tsx

```diff
@@ -10,7 +11,7 @@ const Email = () => {
         setValue(e.currentTarget.value);
     };
 
-    const emailValidation = () => {
+    const emailValidation = async () => {
         const invalidationRules = [
             /^\s*$/, // check string not empty
```

Comment:
Why do you need check "string not empty" if you have check "@ should exist"?

Reply:
The first check is needed for case, when the user clicked on field and then left it (value === '')

Reply:
I understand that string can be empty, but you have check for "@" here. In case of empty string it will not pass checking

Reply:
Unfortunately, the second check not works for empty string, that's why I added special check for it.